test/sqllogictest/audit_log.slt

```diff
@@ -0,0 +1,63 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+# Test expected population of mz_audit_logs after some DDL statements.
+
+mode cockroach
+
+statement ok
+CREATE CLUSTER bar REPLICA r1 (REMOTE ('localhost:1235')), REPLICA r2 (REMOTE ('localhost:1236'))
+
+statement ok
+CREATE MATERIALIZED VIEW v2 AS SELECT 1
+
+statement ok
+CREATE VIEW unmat AS SELECT 1
+
+statement ok
+CREATE TABLE t ()
+
+statement ok
+CREATE DEFAULT INDEX ON t
+
+statement ok
+ALTER VIEW unmat RENAME TO renamed
+
+statement ok
+CREATE OR REPLACE VIEW v2 AS SELECT 2
+
+statement ok
+CREATE DEFAULT INDEX ON renamed
+
+statement ok
+DROP VIEW renamed
+
+statement ok
+CREATE SOURCE s FROM PUBNUB
+SUBSCRIBE KEY 'sub-c-4377ab04-f100-11e3-bffd-02ee2ddab7fe'
+CHANNEL 'pubnub-market-orders';
+
+query TTTT
+SELECT operation, object, op_extra, user FROM mz_audit_log ORDER BY occurred_at
+----
+CreateCluster  bar  null  materialize
+CreateClusterReplica  bar:r1  {"cluster_name":"bar","config":{"Remote":{"replicas":["localhost:1235"]}},"replica_name":"r1"}  materialize
+CreateClusterReplica  bar:r2  {"cluster_name":"bar","config":{"Remote":{"replicas":["localhost:1236"]}},"replica_name":"r2"}  materialize
+CreateView  materialize.public.v2  null  materialize
+CreateIndex  materialize.public.v2_primary_idx  null  materialize
+CreateView  materialize.public.unmat  null  materialize
+CreateIndex  materialize.public.t_primary_idx  null  materialize
+RenameView  materialize.public.unmat  {"new_name":"renamed","previous_name":"materialize.public.unmat"}  materialize
+DropIndex  materialize.public.v2_primary_idx  null  materialize
+DropView  materialize.public.v2  null  materialize
+CreateView  materialize.public.v2  null  materialize
+CreateIndex  materialize.public.renamed_primary_idx  null  materialize
+DropIndex  materialize.public.renamed_primary_idx  null  materialize
+DropView  materialize.public.renamed  null  materialize
+CreateSource  materialize.public.s  null  materialize
```

Comment:
This doesn't quite match the design doc as I understood it! I thought operation was going to be `CREATE`/`DROP`/`ALTER`, object was going to be `cluster`/`cluster_replica`/`index`/`view`, and the name would go in the `op_extra`. That seems a bit cleaner than trying to jam the name into the `object` field, because names have such different structures by object type. E.g. the things that use `db.schema.item` probably want to have three separate fields in the `op_extra` JSON blob.